### PR TITLE
return ReceptionResult from invocation callback

### PR DESCRIPTION
**Pull request by om26er**
Fixes the second half of https://github.com/crossbario/autobahn-java/issues/221

### CODE REVIEW

File: demo-gallery/src/main/java/io/crossbar/autobahn/demogallery/netty/Service.java [hunk 1]
```diff
@@ -314,8 +316,12 @@ public void onJoinHandler4(Session session, SessionDetails details) {
 
 
     // this handler will process incoming events for the topic we subscribe it to
-    private void onCounter(List<Object> args, Map<String, Object> kwargs, EventDetails details) {
+    private CompletableFuture<ReceptionResult> onCounter(List<Object> args,
+                                                      Map<String, Object> kwargs,
+                                                      EventDetails details) {
         System.out.println("received counter: " + args.get(0));
+        CompletableFuture<ReceptionResult> future = new CompletableFuture<>();
```
Comment (om26er) [inline demo-gallery/src/main/java/io/crossbar/autobahn/demogallery/netty/Service.java]: We probably should return a completedFuture here, but we can change this later

**Comment by oberstet** [comment]
The main problem with this change would be: each and every user event handler now needs to construct a CompletableFuture to return, while

1. as of today, the underlying WAMP protocol does NOT yet have the ability to do sth useful with this (see [here](https://github.com/wamp-proto/wamp-proto/issues/280))
2. even if we have 1. in the future, the 95% case of user event handlers would probably NOT want to use acknowledged event delivery, but only regular, un-ack'ed delivery - and hence it would be annoying to write these additional lines of code for users which they dont need

So the question is, do we find a way that allows to cover the 95% case (no ACK needed, nothing returned from the event handler) AND already now can prepare for 1., once EVENT_RECEIVED is spec'ed in WAMP

**Comment by om26er** [comment]
> So the question is, do we find a way that allows to cover the 95% case (no ACK needed, nothing returned from the event handler) AND already now can prepare for 1., once EVENT_RECEIVED is spec'ed in WAMP

sure, please check the latest changes that I pushed.

**Comment by oberstet** [comment]
perfect! cool=)

**Pull request merged**